packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx

```diff
@@ -1051,11 +1056,15 @@ export class ComboBox extends BaseComponent<IComboBoxProps, IComboBoxState> {
           onChange(submitPendingValueEvent, undefined, undefined, currentPendingValue);
         }
       } else {
-        // If we are not controlled, create a new option
+        // If we are not controlled, create a new selected option
         const newOption: IComboBoxOption = {
           key: currentPendingValue || getId(),
           text: this._normalizeToString(currentPendingValue)
         };
+        // If it's multiselect, set selected state to true
+        if (this.props.multiSelect) {
+          newOption.selected = true;
+        }
         const newOptions: IComboBoxOption[] = [...currentOptions, newOption];
```

Comment:
I know this wasn't changed in the PR, but this looks like just a `concat`

Reply:
Really it should probably be slice, then concat. I can do that.

Reply:
Nope, you're right, just concat makes most sense.

Reply:
Yeah, `concat` doesn't mutate the inputs, so `slice` shouldn't be needed